### PR TITLE
 globalprotect-openconnect: 1.4.8 -> 1.4.9

### PR DESCRIPTION
**Pull request by gou4shi1**
###### Description of changes

https://github.com/yuezk/GlobalProtect-openconnect/releases/tag/v1.4.9

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/globalprotect-openconnect/default.nix [hunk 1]
```diff
@@ -1,22 +1,19 @@
...
-  version = "1.4.8";
+  version = "1.4.9";
 
-  src = fetchFromGitHub {
-    owner = "yuezk";
-    repo = "GlobalProtect-openconnect";
-    fetchSubmodules = true;
-    rev = "v${version}";
-    sha256 = "sha256-PQAlGeHVayImKalCNv2SwPcxD0ts4BVSqeo1hKYmnMA=";
+  src = builtins.fetchTarball {
```
Comment (wegank) [inline pkgs/tools/networking/globalprotect-openconnect/default.nix:9]:
```suggestion
  src = fetchurl {
```

File: pkgs/tools/networking/globalprotect-openconnect/default.nix [hunk 2]
```diff
@@ -1,22 +1,19 @@
 { stdenv, lib, fetchFromGitHub
```
Comment (wegank) [inline pkgs/tools/networking/globalprotect-openconnect/default.nix:1]:
```suggestion
{ stdenv, lib, fetchurl
```

File: pkgs/tools/networking/globalprotect-openconnect/default.nix [hunk 3]
```diff
@@ -1,22 +1,19 @@
...
 
-  src = fetchFromGitHub {
-    owner = "yuezk";
-    repo = "GlobalProtect-openconnect";
-    fetchSubmodules = true;
-    rev = "v${version}";
-    sha256 = "sha256-PQAlGeHVayImKalCNv2SwPcxD0ts4BVSqeo1hKYmnMA=";
+  src = fetchurl {
+    url = "https://github.com/yuezk/GlobalProtect-openconnect/releases/download/v${version}/globalprotect-openconnect-${version}.tar.gz";
+    sha256 = "0b7s3gf0gznlpmf9dxfq77254zlxmpajhzzn3scrdrvf413sjl0f";
```
Comment (wegank) [inline pkgs/tools/networking/globalprotect-openconnect/default.nix:11]: This may also need to be updated.

Comment (wegank) [reply pkgs/tools/networking/globalprotect-openconnect/default.nix:11]: Also, what's the reason for using `fetchurl` instead?

Comment (gou4shi1) [reply pkgs/tools/networking/globalprotect-openconnect/default.nix:11]: I failed to fetch [the new submodule](https://github.com/yuezk/GlobalProtect-openconnect/blob/8ca261055096c3a740b096a8d8106028bb82d76f/.gitmodules#L10) in my NixOS, may related to https://github.com/NixOS/nixpkgs/issues/195117

File: pkgs/tools/networking/globalprotect-openconnect/default.nix [hunk 4]
```diff
@@ -1,22 +1,19 @@
...
-  version = "1.4.8";
+  version = "1.4.9";
 
-  src = fetchFromGitHub {
-    owner = "yuezk";
-    repo = "GlobalProtect-openconnect";
-    fetchSubmodules = true;
-    rev = "v${version}";
-    sha256 = "sha256-PQAlGeHVayImKalCNv2SwPcxD0ts4BVSqeo1hKYmnMA=";
+  src = fetchurl {
```
Comment (gou4shi1) [inline pkgs/tools/networking/globalprotect-openconnect/default.nix:9]: FYI
> Also, what's the reason for using fetchurl instead?

> I failed to fetch [the new submodule](https://github.com/yuezk/GlobalProtect-openconnect/blob/8ca261055096c3a740b096a8d8106028bb82d76f/.gitmodules#L10) in my NixOS, may related to https://github.com/NixOS/nixpkgs/issues/195117

**Review by wegank — APPROVED** [review]

**Comment by wegank** [comment]
@gou4shi1 Would you like to be a maintainer of this package?

**Comment by gou4shi1** [comment]
> @gou4shi1 Would you like to be a maintainer of this package?

sure

**Comment by wegank** [comment]
Then you will need to add yourself to [maintainers/maintainer-list.nix](https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix) in a separate commit, and then to this package.

**Comment by jerith666** [comment]
This update _looks_ fine to me, but I was hoping to confirm that I could still connect to my company's VPN before approving the PR.

So, I've been trying to rebuild this along with #212806.  But, the `webkitgtk` rebuilds that result keep getting killed by the `oom-killer`.  :(

Though I see that other PR merged to master overnight, so maybe those builds will be cached now; I'll try again.

**Review by jerith666 — APPROVED** [review]
connects to my corporate VPN fine!

**Pull request merged**